src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int ARRAY_EXTEND_COEFFICIENT = 2;
+
+    private int arrayCapacity = DEFAULT_CAPACITY;
+    private int threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    private int size;
+    private Node[] hashMapArray = new Node[arrayCapacity];
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int elementPosition = findElementPosition(key);
+        if (hashMapArray[elementPosition] == null) {
+            hashMapArray[elementPosition] = new Node(key, value, null);
+            size++;
+            return;
+        }
+        Node node = hashMapArray[elementPosition];
+        while (true) {
+            if (key == null || node.key == null) {
+                if (key == null && node.key == null) {
+                    node.value = value;
+                    return;
+                }
+            } else if (node.key.equals(key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node(key, value, null);
+                size++;
+                return;
+            }
+            node = node.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int elementPosition = findElementPosition(key);
+        Node node = hashMapArray[elementPosition];
+        while (node != null) {
+            if (key == null || node.key == null) {
+                if (key == null && node.key == null) {
+                    return (V) node.value;
+                }
+            } else {
+                if (node.key.equals(key)) {
+                    return (V) node.value;
+                }
+            }
```

Comment:
Please use my example from comment above to implement standard null-safe equals check